### DISCUSSION TITLE
caBundle management

### DISCUSSION THREAD

**Discussion by PSanetra** [kubemod/kubemod#76] [Q&A] [answered]
By looking at the documentation I don't understand how the `caBundle` in the `MutatingWebhookConfiguration` is initialized and rotated. Is it possible to manage it automatically via a `cert-manager`? See https://cert-manager.io/docs/concepts/webhook/

**Comment by vassilvk** [comment, marked as answer]
Out of the box, the TLS certs for KubeMod's webhook are generated at deployment time by a job which uses [kubemod/kubemod-crt](https://github.com/kubemod/kubemod-crt) to generate the signing CA and the webhook cert itself.
The certs are not rotated periodically yet - there is an issue for that: #71.

The main driver behind using kubemod-crt is to keep KubeMod independent of external dependencies such as cert-manager.
That said, since KubeMod itself does not depend on kubemod-crt, one should be able to replace kubemod-crt with an alternative (like cert-manager and ca-injector).
This will obviously require a modified bundle.yaml (to remove kubemod-crt job and replace with cert-manager), but it's doable.
